web_hierarchy_list/static/src/hierarchy_list_renderer.esm.js

```diff
@@ -0,0 +1,108 @@
+import {onWillStart, useState} from "@odoo/owl";
+import {HierarchyListBreadcrumb} from "./hierarchy_list_breadcrumb.esm";
+import {ListRenderer} from "@web/views/list/list_renderer";
+import {evaluateBooleanExpr} from "@web/core/py_js/py";
+import {useSetupAction} from "@web/search/action_hook";
+
+export class HierarchyListRenderer extends ListRenderer {
+    static components = {
+        ...ListRenderer.components,
+        HierarchyListBreadcrumb,
+    };
+    static props = [...ListRenderer.props, "onParentRecordUpdate"];
+    static template = "web_hierarchy_list.HierarchyListRenderer";
+    static rowsTemplate = "web_hierarchy_list.HierarchyListRenderer.Rows";
+    static recordRowTemplate = "web_hierarchy_list.HierarchyListRenderer.RecordRow";
+    setup() {
+        super.setup();
+        useSetupAction({
+            getGlobalState: () => {
+                // We only persist the breadcrumb state in the global state if it was provided
+                // by the environment. Indeed, the environment variable is created by the
+                // HierarchyListController, which ensures that the state is only persisted there
+                // and not when the renderer is used in a x2Many field.
+                if (
+                    !this.env.breadcrumbState ||
+                    this.state.breadcrumbState.length === 0
+                ) {
+                    return {};
+                }
+                return {
+                    breadcrumbState: this._getBreadcrumbState(),
+                };
+            },
+        });
+        // As the breadcrumb state is not provided when the renderer is mounted into a x2Many
+        // field, we need to have a fallback value.
+        this.state = useState({
+            breadcrumbState: this.env.breadcrumbState || [],
+        });
+        onWillStart(this.willStart);
```

Comment:
haha, i was going to comment that this does not work, because this.willStart is unbound, but it actually does, because owl is calling it with the component as `this`. I had forgotten about that